From 3443279cbe60cf44dce40f7ad0fa4eaa6352fddc Mon Sep 17 00:00:00 2001
From: VinceChang6637 <vince_chang@aspeedtech.com>
Date: Tue, 9 Jul 2024 15:19:02 +0800
Subject: [PATCH] fix pfr-manager crash

This issue was introduced by
https://github.com/openbmc/pfr-manager/commit/be13259dd6f1dc2fac52792be091c97d7223596f.

If systemd doesn't start up all of service the FinishTimestamp is zero.
The pfr-manger run monitorSignals at checkPfrInterface() cause process
crash
```
root@ast2600-dcscm:~#  busctl get-property  org.freedesktop.systemd1 \
	/org/freedesktop/systemd1 org.freedesktop.systemd1.Manager \
	FinishTimestamp
t 0

root@ast2600-dcscm:~# journalctl -a
                :
Jul 09 00:36:56 ast2600-dcscm systemd[1]:
xyz.openbmc_project.PFR.Manager.service: Main process exited,
code=dumped, status=11/SEGV
Jul 09 00:36:56 ast2600-dcscm systemd[1]:
xyz.openbmc_project.PFR.Manager.service: Failed with result 'core-dump'.
```

Signed-off-by: VinceChang6637 <vince_chang@aspeedtech.com>
---
 service/src/mainapp.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/service/src/mainapp.cpp b/service/src/mainapp.cpp
index b913e33..3bcbff1 100644
--- a/service/src/mainapp.cpp
+++ b/service/src/mainapp.cpp
@@ -628,7 +628,9 @@ void checkPfrInterface(std::shared_ptr<sdbusplus::asio::connection> conn,
         else
         {
             unProvChkPointStatus = true;
-            pfr::monitorSignals(server, conn);
+            // Mask monitorSignals to avoid pfr-manager crash, if systemd
+            // FinishTimestamp is zero. 
+            //pfr::monitorSignals(server, conn);
         }
     }
 }
